src/codegen/c.zig

```diff
@@ -420,8 +422,12 @@ pub const DeclGen = struct {
         const int_info = @typeInfo(@TypeOf(int_val)).Int;
         const is_signed = int_info.signedness == .signed;
         const is_neg = int_val < 0;
+
         comptime assert(int_info.bits > 64 and int_info.bits <= 128);
 
+        // TODO support big endian arch
+        comptime assert(builtin.cpu.arch.endian() == .Little);
```

Comment:
This is checking the endianness of the compiler, not the target. Also can you just fix the bug, if any, rather than asserting? It's not that hard to solve.